### PR TITLE
Cash sniffing 2.0

### PR DESCRIPTION
**Pull request by pudl**
:cl: Quiltyquilty
rscadd: Clicking on space cash while it's in your hand will now make you sniff it.
/:cl:

This adds cash sniffing again, but it's improved on the last pr in several ways.
- The QM no longer has his clipboard stolen.
- The QM isn't given a lot of money, that was unbalanced.
- There is now a sound effect when you sniff the cash.

Thanks everyone.

### CODE REVIEW

**Comment by X-TheDark** [comment]
Now just implement being able to add chems on space cash to PUNISH those who sniff it and you're golden.

**Comment by PKPenguin321** [comment]
[Link to listen to sound in browser](http://puu.sh/pyqCN/ed2b1dd8cc.mp3)

**Comment by Cheridan** [comment]
Uh this is not immersive enough, you have to add checks to see if equipment covering your face and if it is then you can't sniff

**Comment by MMMiracles** [comment]
oh my god what is that sniff sound its like a baby elephant

**Comment by lzimann** [comment]
what the actual fuck who the fuck sniffs cash

**Comment by YakumoChen** [comment]
Good meme

File: code/game/objects/items/stacks/cash.dm [hunk 1]
```diff
@@ -12,6 +12,15 @@
 	w_class = 1
 	burn_state = FLAMMABLE
 	var/value = 1
+	var/cooldown = 0
+
+/obj/item/stack/spacecash/attack_self(mob/user as mob)
+	if(user.r_hand == src || user.l_hand == src)
+		if(cooldown < world.time - 15)
```
Comment (KorPhaeron) [inline code/game/objects/items/stacks/cash.dm]: This is 1.5 seconds, still seems very spammable

**Comment by pudl** [comment]
Ugh it doesnt compile

File: code/game/objects/items/stacks/cash.dm [hunk 2]
```diff
@@ -12,6 +12,19 @@
 	w_class = 1
 	burn_state = FLAMMABLE
 	var/value = 1
+	var/cooldown = 0
+
+/obj/item/stack/spacecash/attack_self(mob/user as mob)
```
Comment (RemieRichards) [inline code/game/objects/items/stacks/cash.dm]: No ```as X``` in proc args

File: code/game/objects/items/stacks/cash.dm [hunk 3]
```diff
@@ -12,6 +12,19 @@
...
+	if(!canconsume(M, user))
+		return 0
```
Comment (X-TheDark) [inline code/game/objects/items/stacks/cash.dm]: What is this check supposed to do? Also, what's this `M` and where does it come from? (Copy-pasting code without checking it afterwards is bad).

`canconsume(mob/eater, mob/user)` is defined for items of type `/obj/item/weapon/reagent_containers/` as per https://github.com/tgstation/tgstation/blob/master/code/modules/reagents/reagent_containers.dm#L55

**Pull request closed without merging**